src/CMakeLists.txt

```diff
@@ -37,6 +37,14 @@ add_library(
   ${${PROJECT_NAME}_SOURCE_DIR}/include/parsers/qasm_parser/Token.hpp
   ${${PROJECT_NAME}_SOURCE_DIR}/include/Permutation.hpp
   ${${PROJECT_NAME}_SOURCE_DIR}/include/QuantumComputation.hpp
+  ${${PROJECT_NAME}_SOURCE_DIR}/include/ecc/Ecc.hpp
+  ${${PROJECT_NAME}_SOURCE_DIR}/include/ecc/Id.hpp
+  ${${PROJECT_NAME}_SOURCE_DIR}/include/ecc/Q3Shor.hpp
+  ${${PROJECT_NAME}_SOURCE_DIR}/include/ecc/Q5Laflamme.hpp
+  ${${PROJECT_NAME}_SOURCE_DIR}/include/ecc/Q7Steane.hpp
+  ${${PROJECT_NAME}_SOURCE_DIR}/include/ecc/Q9Shor.hpp
+  ${${PROJECT_NAME}_SOURCE_DIR}/include/ecc/Q9Surface.hpp
+  ${${PROJECT_NAME}_SOURCE_DIR}/include/ecc/Q18Surface.hpp
```

Comment:
One thing I failed to mention earlier: these files should all be in a separate CMake target. Not everyone should be forced to build these ECC features if they are not using them.
This can be accomplished in a similar fashion as the `qfr_dd` and `qfr_zx` targets.
In addition, the tests should be in their own target that links against the newly introduced target and the DD `qfr_dd` target.

Reply:
Done